WordPress/openverse
# Add an updated `VAudioControl` component

## Fixes



Fixes #2602 by @fcoveram

## Description


This PR creates a new `VAudioControl` component that is based on the `VPlayPause` button to match the [Audio control](https://www.figma.com/file/GIIQ4sDbaToCfFQyKMvzr8/Openverse-Design-Library?type=design&node-id=5215-31484&mode=dev) component in Design Library.

This component is used in the `VAudioControl` story in the storybook. It is **not** used in the audio track components because that would require updates to the audio track layouts.

To implement the updated styles, the following changes were added:
- `larger` size that is 64x64px (w-16 h-16) - for the large VPlayPause component
- `connections` prop to the `VIconButton` to allow for adding connections (non-rounded corners on the side where the play-pause button is joined to the waveform, for example)

The Audio track components are very complicated, they use dynamic components with `is` and several layers of scoped slots. This breaks the type-safety of the props that are passed the `VAudioControl`. To make `VAudioControl` as a child of a dynamic component that is used within a scoped slot, safer, I added `validator`s to its props. They cause errors at runtime if the props are invalid even if TypeScript does not have any errors.

## Testing Instructions

Check that the new Audio control matches the design requirements using the rendered story: https://docs.openverse.org/_preview/2929/storybook/?path=/story/components-audio-track-audio-control--default-story 

Check various states (resting/ hover/ focus).

@fcoveram, this component does not have an `active` state anymore, so there is not indication of when you click it, as we used to have (the white ring that you can see in prod). Should we have something for that?

## Checklist



[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
